### PR TITLE
nss: 3.75, separate nspr; nspr: new at 4.33

### PR DESCRIPTION
**Pull request by ioraff**
I've separated these so Firefox can be built with the system nss and nspr. Chromium also builds with this setup, but qt5-webengine complains about python not being present (when it is). Not sure how to fix this.

## Installed manifest of package

nss

```
/var/db/kiss/installed/nss/version
/var/db/kiss/installed/nss/sources
/var/db/kiss/installed/nss/manifest
/var/db/kiss/installed/nss/depends
/var/db/kiss/installed/nss/checksums
/var/db/kiss/installed/nss/build
/var/db/kiss/installed/nss/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/lib/pkgconfig/nss.pc
/usr/lib/pkgconfig/
/usr/lib/libssl3.so
/usr/lib/libsoftokn3.so
/usr/lib/libsoftokn3.chk
/usr/lib/libsmime3.so
/usr/lib/libpkcs11testmodule.so
/usr/lib/libnssutil3.so
/usr/lib/libnsssysinit.so
/usr/lib/libnssdbm3.so
/usr/lib/libnssdbm3.chk
/usr/lib/libnssckbi.so
/usr/lib/libnssckbi-testlib.so
/usr/lib/libnss3.so
/usr/lib/libfreeblpriv3.so
/usr/lib/libfreeblpriv3.chk
/usr/lib/libfreebl3.so
/usr/lib/libfreebl3.chk
/usr/lib/
/usr/include/nss/utilrename.h
/usr/include/nss/utilparst.h
/usr/include/nss/utilpars.h
/usr/include/nss/utilmodt.h
/usr/include/nss/sslt.h
/usr/include/nss/sslproto.h
/usr/include/nss/sslexp.h
/usr/include/nss/sslerr.h
/usr/include/nss/ssl.h
/usr/include/nss/smime.h
/usr/include/nss/shsign.h
/usr/include/nss/secport.h
/usr/include/nss/secpkcs7.h
/usr/include/nss/secpkcs5.h
/usr/include/nss/secoidt.h
/usr/include/nss/secoid.h
/usr/include/nss/secmodt.h
/usr/include/nss/secmod.h
/usr/include/nss/secmime.h
/usr/include/nss/secitem.h
/usr/include/nss/sechash.h
/usr/include/nss/secerr.h
/usr/include/nss/secdigt.h
/usr/include/nss/secdig.h
/usr/include/nss/secdert.h
/usr/include/nss/secder.h
/usr/include/nss/seccomon.h
/usr/include/nss/secasn1t.h
/usr/include/nss/secasn1.h
/usr/include/nss/private/blapi.h
/usr/include/nss/private/alghmac.h
/usr/include/nss/private/
/usr/include/nss/preenc.h
/usr/include/nss/portreg.h
/usr/include/nss/pkcs7t.h
/usr/include/nss/pkcs1sig.h
/usr/include/nss/pkcs12t.h
/usr/include/nss/pkcs12.h
/usr/include/nss/pkcs11uri.h
/usr/include/nss/pkcs11u.h
/usr/include/nss/pkcs11t.h
/usr/include/nss/pkcs11p.h
/usr/include/nss/pkcs11n.h
/usr/include/nss/pkcs11f.h
/usr/include/nss/pkcs11.h
/usr/include/nss/pk11sdr.h
/usr/include/nss/pk11pub.h
/usr/include/nss/pk11priv.h
/usr/include/nss/pk11pqg.h
/usr/include/nss/pk11hpke.h
/usr/include/nss/pk11func.h
/usr/include/nss/p12t.h
/usr/include/nss/p12plcy.h
/usr/include/nss/p12.h
/usr/include/nss/ocspt.h
/usr/include/nss/ocsp.h
/usr/include/nss/nssutil.h
/usr/include/nss/nssrwlkt.h
/usr/include/nss/nssrwlk.h
/usr/include/nss/nsslowhash.h
/usr/include/nss/nsslocks.h
/usr/include/nss/nssilock.h
/usr/include/nss/nssilckt.h
/usr/include/nss/nssckt.h
/usr/include/nss/nssckmdt.h
/usr/include/nss/nssckg.h
/usr/include/nss/nssckfwt.h
/usr/include/nss/nssckfwc.h
/usr/include/nss/nssckfw.h
/usr/include/nss/nssckft.h
/usr/include/nss/nssckepv.h
/usr/include/nss/nssckbi.h
/usr/include/nss/nssbaset.h
/usr/include/nss/nssbase.h
/usr/include/nss/nssb64t.h
/usr/include/nss/nssb64.h
/usr/include/nss/nss.h
/usr/include/nss/lowkeyti.h
/usr/include/nss/lowkeyi.h
/usr/include/nss/keythi.h
/usr/include/nss/keyt.h
/usr/include/nss/keyhi.h
/usr/include/nss/key.h
/usr/include/nss/jarfile.h
/usr/include/nss/jar.h
/usr/include/nss/jar-ds.h
/usr/include/nss/hasht.h
/usr/include/nss/ecl-exp.h
/usr/include/nss/eccutil.h
/usr/include/nss/cryptoht.h
/usr/include/nss/cryptohi.h
/usr/include/nss/crmft.h
/usr/include/nss/crmf.h
/usr/include/nss/cmst.h
/usr/include/nss/cmsreclist.h
/usr/include/nss/cms.h
/usr/include/nss/cmmft.h
/usr/include/nss/cmmf.h
/usr/include/nss/ciferfam.h
/usr/include/nss/certt.h
/usr/include/nss/certdb.h
/usr/include/nss/cert.h
/usr/include/nss/blapit.h
/usr/include/nss/base64.h
/usr/include/nss/
/usr/include/
/usr/
```

nspr
```
/var/db/kiss/installed/nspr/version
/var/db/kiss/installed/nspr/sources
/var/db/kiss/installed/nspr/manifest
/var/db/kiss/installed/nspr/checksums
/var/db/kiss/installed/nspr/build
/var/db/kiss/installed/nspr/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/lib/pkgconfig/nspr.pc
/usr/lib/pkgconfig/
/usr/lib/libplds4.so
/usr/lib/libplc4.so
/usr/lib/libnspr4.so
/usr/lib/
/usr/include/nspr/prwin16.h
/usr/include/nspr/prvrsion.h
/usr/include/nspr/prtypes.h
/usr/include/nspr/prtrace.h
/usr/include/nspr/prtpool.h
/usr/include/nspr/prtime.h
/usr/include/nspr/prthread.h
/usr/include/nspr/prsystem.h
/usr/include/nspr/prshma.h
/usr/include/nspr/prshm.h
/usr/include/nspr/prrwlock.h
/usr/include/nspr/prrng.h
/usr/include/nspr/prproces.h
/usr/include/nspr/prprf.h
/usr/include/nspr/prpdce.h
/usr/include/nspr/prolock.h
/usr/include/nspr/prnetdb.h
/usr/include/nspr/prmwait.h
/usr/include/nspr/prmon.h
/usr/include/nspr/prmem.h
/usr/include/nspr/prlong.h
/usr/include/nspr/prlog.h
/usr/include/nspr/prlock.h
/usr/include/nspr/prlink.h
/usr/include/nspr/private/prpriv.h
/usr/include/nspr/private/pprthred.h
/usr/include/nspr/private/pprio.h
/usr/include/nspr/private/
/usr/include/nspr/pripcsem.h
/usr/include/nspr/prio.h
/usr/include/nspr/prinrval.h
/usr/include/nspr/prinit.h
/usr/include/nspr/prinet.h
/usr/include/nspr/prerror.h
/usr/include/nspr/prerr.h
/usr/include/nspr/prenv.h
/usr/include/nspr/prdtoa.h
/usr/include/nspr/prcvar.h
/usr/include/nspr/prcpucfg.h
/usr/include/nspr/prcountr.h
/usr/include/nspr/prcmon.h
/usr/include/nspr/prclist.h
/usr/include/nspr/prbit.h
/usr/include/nspr/pratom.h
/usr/include/nspr/plstr.h
/usr/include/nspr/plhash.h
/usr/include/nspr/plgetopt.h
/usr/include/nspr/plerror.h
/usr/include/nspr/plbase64.h
/usr/include/nspr/plarenas.h
/usr/include/nspr/plarena.h
/usr/include/nspr/obsolete/prsem.h
/usr/include/nspr/obsolete/protypes.h
/usr/include/nspr/obsolete/probslet.h
/usr/include/nspr/obsolete/pralarm.h
/usr/include/nspr/obsolete/
/usr/include/nspr/nspr.h
/usr/include/nspr/
/usr/include/
/usr/bin/nspr-config
/usr/bin/
/usr/
```

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify the @kiss-community maintainers if I can no longer maintain this package.

## Existing package

- [x] I am the maintainer of this package.

### CODE REVIEW

**Comment by dilyn-corner** [comment]
Curious to see that we engine error

Jan 13, 2022, 3:25 PM by ***@***.***:

**Comment by ioraff** [comment]
https://0x0.st/o-_-.txt

**Comment by dilyn-corner** [comment]
```
nss .................................. yes
```

Seems to be detected just fine?
```
  QtWebEngine required system libraries for qpa-xcb:
    x11 .................................. yes
    libdrm ............................... yes
    xcomposite ........................... yes
    xcursor .............................. yes
    xi ................................... yes
    xproto (glproto) ..................... yes
    xtst ................................. yes
    xkbfile .............................. yes
```

I see I see



I'd hazard a guess and say that the `python3` patch doesn't actually accomplishes its goals somehow, any more. The build system for `qt5-webengine` is held up precariously by some paper thin sticks... I'll see if I can figure this out.

**Comment by ioraff** [comment]
Tried building with the python from the main repo--in case the libressl patch broke something--to no avail. Are you not having the same problem?

**Comment by dilyn-corner** [comment]
I don't use QT anymore, but this latest version had built just fine for me in a chroot before I pushed the update.

**Comment by dilyn-corner** [comment]
Try this python patch:
http://ix.io/3OAg

**Comment by ioraff** [comment]
Python detection is fixed, but the build failed. Looks like it's due to ffmpeg 5.0. Trying again with 4.4.1.

**Comment by dilyn-corner** [comment]
> Looks like it's due to ffmpeg 5.0

Gross. I'll look into that as well.

**Comment by ioraff** [comment]
Okay. Well, can this be merged since nss isn't the limiting factor?

**Comment by dilyn-corner** [comment]
The real question is why you want to build `firefox` with the system `nss`?

**Comment by ioraff** [comment]
I'd rather a security-oriented library not be vendored, and I occasionally use chromium, so there's no point in having a vendored nss if it's going to be installed anyway.

File: community/nss/build [hunk 1]
```diff
@@ -4,46 +4,35 @@ export NSS_USE_SYSTEM_SQLITE=1
...
 # Yucky. This is needed by qtwebengine.
-install -Dt "$1/usr/include/nss/obsolete" \
-    -m644 dist/Linux*/include/obsolete/*.h
-
-# Install the NSPR files. This is a joint package as I'd
-# rather not juggle which version of NSPR works best with NSS.
-install -Dt "$1/usr/include/nspr" -m644 nspr/pr/include/*.h
-install -Dt "$1/usr/include/nspr" -m644 nspr/lib/*/*.h
+#install -Dt "$1/usr/include/nss/obsolete" \
+#    -m644 dist/Linux*/include/obsolete/*.h
```
Comment (dilyn-corner) [inline community/nss/build:25]: Probably don't need these commented, but last I checked webengine DID need these headers.

File: community/nspr/build [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    --enable-optimize \
+    --enable-64bit
+
+make
+make install
+
+rm -f -- "$1"/usr/lib/*.a
+rm -rf "$1"/usr/bin/prerr.properties \
+    "$1"/usr/bin/compile-et.pl \
+    "${1:?}"/usr/share \
```
Comment (dilyn-corner) [inline community/nspr/build:19]: $1 should never be null or unset

Comment (dilyn-corner) [inline community/nspr/build:19]: All of these should be quoted as something like...

```
rm -f "$1/usr/lib/"*.a
rm -rf "$1/usr/bin/prerr.properties"
```
etc.

File: community/nspr/build [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+
+mkdir build inst
+
+cd build
+CFLAGS="$CFLAGS -D_PR_POLL_AVAILABLE -D_PR_HAVE_OFF64_T -D_PR_INET6 -D_PR_HAVE_INET_NTOP -D_PR_HAVE_GETHOSTBYNAME2 -D_PR_HAVE_GETADDRINFO -D_PR_INET6_PROBE" \
```
Comment (dilyn-corner) [inline community/nspr/build:6]: Are these flags necessary?

**Review by dilyn-corner — CHANGES_REQUESTED** [review]
🤔 

I suppose we can do this, I don't see any technical reason it can't be done.

File: community/nspr/build (hunk 3 above)
Comment (ioraff) [reply community/nspr/build:6]: https://git.alpinelinux.org/aports/tree/main/nspr/APKBUILD#n30

File: community/nss/build (hunk 1 above)
Comment (ioraff) [reply community/nss/build:25]: I was just able to build successfully without them

File: community/nspr/build (hunk 3 above)
Comment (dilyn-corner) [reply community/nspr/build:6]: Of course it wouldn't be smart about musl...

File: community/nspr/build [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+../nspr/configure \
+    --prefix=/usr \
+    --disable-debug \
+    --enable-optimize \
+    --enable-64bit
+
+make
+make install
+
+# $1 is never unset
```
Comment (dilyn-corner) [inline community/nspr/build:16]: This is very interesting to me though.

There are sixteen instances of `rm -rf "$1/..."` in the main repository, but only *one* (`expat`) includes a check of `${1:?}` 🤔 Just an observation.

File: community/nss/build (hunk 1 above)
Comment (dilyn-corner) [reply community/nss/build:25]: Then we're safe to remove these lines completely!

File: community/nspr/build (hunk 4 above)
Comment (ioraff) [reply community/nspr/build:18]: I'd guess shellcheck only cares about `/{usr/{bin,lib,share}}` to avoid breaking the system.

Comment (dilyn-corner) [reply community/nspr/build:18]: Quite possible...

**Pull request merged**